Wrong ternary suggested

### Bug description

```python
# pylint: disable=missing-module-docstring,invalid-name

a = True
b = False
c = True
x = (a and b) or c
print(x)  #  x == True
print(b if a else c)  #  ternary suggested == False
```


### Configuration

_No response_

### Command used

```shell
pylint test.py
```


### Pylint output

```shell
************* Module test
test.py:6:0: R1706: Consider using ternary (b if a else c) (consider-using-ternary)
```


### Expected behavior

Success.

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.0
Python 3.9.7 (default, Aug 30 2021, 00:00:00) 
[GCC 11.2.1 20210728 (Red Hat 11.2.1-1)]
```


### OS / Environment

Fedora 34

### Additional dependencies

_No response_

--- a/ChangeLog
+++ b/ChangeLog
@@ -11,6 +11,11 @@ Release date: TBA
 ..
   Put new features here and also in 'doc/whatsnew/2.12.rst'
 
+* Fix proposing ``consider-using-ternary`` incorrectly over ``simplify-boolean-expression``
+  when the condition can be inferred as False
+
+  Closes #5200
+
 * Fix exception when pyreverse parses ``property function`` of a class.
 
 * Add an optional extension ``consider-using-any-or-all`` : Emitted when a ``for`` loop only

--- a/ChangeLog
+++ b/ChangeLog
@@ -11,8 +11,7 @@ Release date: TBA
 ..
   Put new features here and also in 'doc/whatsnew/2.12.rst'
 
-* Fix proposing ``consider-using-ternary`` incorrectly over ``simplify-boolean-expression``
-  when the condition can be inferred as False
+* Fix ``simplify-boolean-expression`` when condition can be inferred as False.
 
   Closes #5200
 

--- a/doc/whatsnew/2.12.rst
+++ b/doc/whatsnew/2.12.rst
@@ -64,6 +64,11 @@ Extensions
 Other Changes
 =============
 
+* Fix proposing ``consider-using-ternary`` incorrectly over ``simplify-boolean-expression``
+  when the condition can be inferred as False
+
+  Closes #5200
+
 * Fix exception when pyreverse parses ``property function`` of a class.
 
 * Improve and flatten ``unused-wildcard-import`` message

--- a/doc/whatsnew/2.12.rst
+++ b/doc/whatsnew/2.12.rst
@@ -64,8 +64,7 @@ Extensions
 Other Changes
 =============
 
-* Fix proposing ``consider-using-ternary`` incorrectly over ``simplify-boolean-expression``
-  when the condition can be inferred as False
+* Fix ``simplify-boolean-expression`` when condition can be inferred as False.
 
   Closes #5200
 

--- a/pylint/checkers/refactoring/refactoring_checker.py
+++ b/pylint/checkers/refactoring/refactoring_checker.py
@@ -1393,10 +1393,10 @@ class RefactoringChecker(checkers.BaseTokenChecker):
             return
 
         inferred_truth_value = utils.safe_infer(truth_value)
-        if inferred_truth_value in (None, astroid.Uninferable):
+        if inferred_truth_value is None or inferred_truth_value == astroid.Uninferable:
             truth_boolean_value = True
         else:
-            truth_boolean_value = inferred_truth_value.bool_value()  # type: ignore
+            truth_boolean_value = inferred_truth_value.bool_value()
 
         if truth_boolean_value is False:
             message = "simplify-boolean-expression"

--- a/pylint/checkers/refactoring/refactoring_checker.py
+++ b/pylint/checkers/refactoring/refactoring_checker.py
@@ -1396,7 +1396,7 @@ class RefactoringChecker(checkers.BaseTokenChecker):
         if inferred_truth_value in (None, astroid.Uninferable):
             truth_boolean_value = True
         else:
-            truth_boolean_value = truth_value.bool_value()
+            truth_boolean_value = inferred_truth_value.bool_value()  # type: ignore
 
         if truth_boolean_value is False:
             message = "simplify-boolean-expression"

--- a/tests/functional/t/ternary.py
+++ b/tests/functional/t/ternary.py
@@ -31,3 +31,9 @@ def func4():
     """"Using a Name as a condition but still emits"""
     truth_value = 42
     return condition and truth_value or false_value # [consider-using-ternary]
+
+
+def func5():
+    """"Using a Name that infers to False as a condition does not emit"""
+    falsy_value = False
+    return condition and falsy_value or false_value # [simplify-boolean-expression]

--- a/tests/functional/t/ternary.txt
+++ b/tests/functional/t/ternary.txt
@@ -5,3 +5,4 @@ consider-using-ternary:19:0::Consider using ternary ('greater' if SOME_VALUE1 > 
 consider-using-ternary:20:0::Consider using ternary ('both' if SOME_VALUE2 > 4 and SOME_VALUE3 else 'not')
 simplify-boolean-expression:23:0::Boolean expression may be simplified to SOME_VALUE2
 consider-using-ternary:33:4:func4:Consider using ternary (truth_value if condition else false_value)
+simplify-boolean-expression:39:4:func5:Boolean expression may be simplified to false_value

